configfileparser.cpp

```diff
@@ -93,6 +93,27 @@ bool ConfigFileParser::testKeyValidity(const std::string &key, const std::string
     {
         std::ostringstream oss;
         oss << "Config key '" << key << "' is not valid (here).";
+
+        // Maybe the user made a typo
+        const std::string *alternative = nullptr;
```

Comment:
Can you instead of pointers use iterators? The empty case can be `validKeys.end()`, which is one after the range. Testing for `!= end()` is idiomatic after a search.

Reply:
Absolutely: done.